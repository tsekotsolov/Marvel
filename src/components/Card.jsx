import React from 'react'
import {Link} from 'react-router-dom'

class Card extends React.Component{

  toggleFavourites=()=>{  
    
    let action;
    if(this.props.appState.favArray.includes(this.props.id)){
      action='remove'
    }else{
      action='add'
    }
    this.props.manageFavArray(this.props.id,action)
  }

  render(){

    return (
      <React.Fragment>
        <div className='col-md-3 col-sm-4 col-xs-12'>
          <div className='card'>
            <div className='card-body'>
            {this.props.name?<p className='card-text'>{`${this.props.name}`}</p>:<p className='card-text'>{`${this.props.title}`}</p>}
            </div>
            <img className='card-img-top' src={`${this.props.thumbnail.path}/portrait_fantastic.${this.props.thumbnail.extension}`} alt='Card cap' />
            {this.props.name?
              <div className='link-wrapper'>
              <Link to={`/character-details?${this.props.id}`}>Details</Link>
              <Link to='#' onClick={this.toggleFavourites}>{
                this.props.appState.favArray.includes(this.props.id)?
                <i className="fas fa-heart"></i>:
                <i className="far fa-heart"></i>}
              </Link>
              </div>:
            null}
          </div>
        </div>
      </React.Fragment>
    )
  }
}

export default Card
