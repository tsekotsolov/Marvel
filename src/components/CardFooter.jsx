import React from 'react'
import {Link} from 'react-router-dom'

export default function CardFooter() {

    let toggleFavourites=()=>{  

        const {favArray} = this.props.appState
        
        let action;
        if(favArray.includes(this.props.id)){
          action='remove'
        }else{
          action='add'
        }
        this.props.manageFavArray(this.props.id,action)
      }
    
  return (
    <div>
      <div className='link-wrapper'>
              <Link to={`/character-details?${this.props.id}`}>Details</Link>
              <Link to='#' onClick={this.toggleFavourites}>{
                this.props.appState.favArray.includes(this.props.id)?
                <i className="fas fa-heart"></i>:
                <i className="far fa-heart"></i>}
              </Link>
              </div>
    </div>
  )
}
