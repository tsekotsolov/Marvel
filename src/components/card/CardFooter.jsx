import React from "react";
import { connect } from "react-redux";
import addRemoveFavourites from "../../actions/addRemoveFavourites";
import StyledCard from "./StyledCard";
import { Link } from "react-router-dom";

const mapstateToProps = state => ({
  favArray: state.favArray
});

const CardFooter = props => {
  const toggleFavourites = () => {
    props.dispatch(addRemoveFavourites(props.id));
    if (props.filteredArray) {
      const filtered = props.filteredArray.filter(
        character => character.id !== props.id
      );
      props.modifyFilteredArray(filtered, true);
    }
  };

  return (
    <StyledCard.Footer>
      <Link to={`/character-details?${props.id}`}>
        <StyledCard.Link>Details</StyledCard.Link>
      </Link>

      <StyledCard.Link onClick={toggleFavourites}>
        {props.favArray.includes(props.id) ? (
          <i className="fas fa-heart" />
        ) : (
          <i className="far fa-heart" />
        )}
      </StyledCard.Link>
    </StyledCard.Footer>
  );
};
export default connect(mapstateToProps)(CardFooter);
