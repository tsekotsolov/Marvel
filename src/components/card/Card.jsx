import React from "react";
import CardFooter from "./CardFooter";
import StyledCard from "./StyledCard";
import Img from "./Img";


const Card = props => {
  let { name, title, thumbnail } = props;

  return (
    <div className="col-md-3 col-sm-4 col-xs-12">
      <div className="card m-2">
        <div className="card-body">
          {name ? (
            <StyledCard.Paragraph>{`${name}`}</StyledCard.Paragraph>
          ) : (
            <StyledCard.Paragraph>{`${title}`}</StyledCard.Paragraph>
          )}
        </div>

        <Img
          path={`${thumbnail.path}/standard_fantastic.${thumbnail.extension}`}
        />

        {name && (
          <CardFooter
            {...props}
            filteredArray={props.filteredArray}
            modifyFilteredArray={props.modifyFilteredArray}
          />
        )}
      </div>
    </div>
  );
};
export default Card;
