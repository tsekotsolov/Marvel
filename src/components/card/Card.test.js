import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Card from "./Card";
import CardFooter from "./CardFooter";

configure({ adapter: new Adapter() });

describe("<CardFooter />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Card name path thumbnail />);
  });

  // it("should have <CardFooter /> if there is name prop> ", () => {
  //   expect(wrapper.find(CardFooter)).toHaveLength(0);
  // });

  // it("should have one <p> tag if there is name prop> ", () => {
  //   expect(wrapper.find("p")).toHaveLength(1);
  // });

  // it("should have one <p> tag if there is title prop> ", () => {
  //   wrapper.setProps({ title: true, name: false });
  //   expect(wrapper.find("p")).toHaveLength(1);
  // });
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
