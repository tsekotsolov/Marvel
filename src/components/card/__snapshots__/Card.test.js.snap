// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFooter /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    name={true}
    path={true}
    thumbnail={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card m-2"
      >
        <div
          className="card-body"
        >
          <ForwardRef>
            true
          </ForwardRef>
        </div>
        <UNDEFINED
          fallback={<Loading />}
        >
          <UNDEFINED
            path="undefined/standard_fantastic.undefined"
          />
        </UNDEFINED>
        <Connect(CardFooter)
          name={true}
          path={true}
          thumbnail={true}
        />
      </div>,
      "className": "col-md-3 col-sm-4 col-xs-12",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-body"
          >
            <ForwardRef>
              true
            </ForwardRef>
          </div>,
          <UNDEFINED
            fallback={<Loading />}
          >
            <UNDEFINED
              path="undefined/standard_fantastic.undefined"
            />
          </UNDEFINED>,
          <Connect(CardFooter)
            name={true}
            path={true}
            thumbnail={true}
          />,
        ],
        "className": "card m-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef>
              true
            </ForwardRef>,
            "className": "card-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "true",
            },
            "ref": null,
            "rendered": "true",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  color: gray;
  margin: 0;
  font-size: 0.85rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    white-space: wrap;
    overflow: visible;
    cursor: pointer;
  }
",
                ],
              },
              "displayName": "styled.p",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <UNDEFINED
              path="undefined/standard_fantastic.undefined"
            />,
            "fallback": <Loading />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "path": "undefined/standard_fantastic.undefined",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            },
          },
          "type": Symbol(react.suspense),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filteredArray": undefined,
            "modifyFilteredArray": undefined,
            "name": true,
            "path": true,
            "thumbnail": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card m-2"
        >
          <div
            className="card-body"
          >
            <ForwardRef>
              true
            </ForwardRef>
          </div>
          <UNDEFINED
            fallback={<Loading />}
          >
            <UNDEFINED
              path="undefined/standard_fantastic.undefined"
            />
          </UNDEFINED>
          <Connect(CardFooter)
            name={true}
            path={true}
            thumbnail={true}
          />
        </div>,
        "className": "col-md-3 col-sm-4 col-xs-12",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-body"
            >
              <ForwardRef>
                true
              </ForwardRef>
            </div>,
            <UNDEFINED
              fallback={<Loading />}
            >
              <UNDEFINED
                path="undefined/standard_fantastic.undefined"
              />
            </UNDEFINED>,
            <Connect(CardFooter)
              name={true}
              path={true}
              thumbnail={true}
            />,
          ],
          "className": "card m-2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForwardRef>
                true
              </ForwardRef>,
              "className": "card-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "true",
              },
              "ref": null,
              "rendered": "true",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "rules": Array [
                    "
  color: gray;
  margin: 0;
  font-size: 0.85rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    white-space: wrap;
    overflow: visible;
    cursor: pointer;
  }
",
                  ],
                },
                "displayName": "styled.p",
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <UNDEFINED
                path="undefined/standard_fantastic.undefined"
              />,
              "fallback": <Loading />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "path": "undefined/standard_fantastic.undefined",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              },
            },
            "type": Symbol(react.suspense),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filteredArray": undefined,
              "modifyFilteredArray": undefined,
              "name": true,
              "path": true,
              "thumbnail": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
